inference-engine/src/vpu/graph_transformer/src/model/model.cpp

```diff
@@ -1854,6 +1886,21 @@ void ModelObj::disconnectStage(const Stage& stage) {
     //
 
     for (const auto& outEdge : stage->_outputEdges) {
+        // Disconnect from dependency
+        if (const auto& dataToShapeEdge = outEdge->output()->parentDataToShapeEdge()) {
+            const auto& shape = dataToShapeEdge->parent();
+            const auto& dependentStagesEdges = shape->dependentStagesEdges();
+
+            const auto it = std::find_if(dependentStagesEdges.begin(), dependentStagesEdges.end(), [&stage](const StageDependency &edge) {
+                return edge->dependentStage() == stage;
+            });
+
+            if (it != dependentStagesEdges.end()) {
+                const auto edge = *it;
+                removeStageDependency(edge);
+            }
+        }
```

Comment:
I think it is internal work. We remove stage dependency in `disconnectDatas` method for objective reasons and here because the dependent stage is being removed.
So far I do not see cases when it can be useful somewhere else

Reply:
It can be useful here, because it would reasonable named method instead of plain code)

Reply:
New method `void removeStageDependency(const Stage& stage, const Data& parent)` was implemented